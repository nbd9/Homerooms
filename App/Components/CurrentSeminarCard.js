import React from 'react'
import { Text, View } from 'react-native'
import { Avatar } from 'react-native-elements'
import Styles from './Styles/CurrentSeminarsStyles'

const CurrentSeminarCard = ({ day, teacher, onPress }) => {
  console.log(teacher)
  return (
    <View style={Styles.cardView} onPress={onPress}>
      {('picture' in teacher)
        ? <Avatar
          size='large'
          rounded
          source={{ uri: teacher.picture }}
        />
        : <Avatar
          size='large'
          rounded
          title={teacher.name.split(' ').map((n) => n[0]).join('')}
        />
      }

      <View style={Styles.textView}>
        <Text style={Styles.header}>{`"${day}" Days`}</Text>
        <Text style={Styles.body}>Youâ€™re currently scheduled with <Text style={Styles.bodyImportant}>{teacher.name}</Text> on all <Text style={Styles.bodyImportant}>{day} Days</Text>. To make request another teacher, search for one!</Text>
      </View>
    </View>
  )
}

export default CurrentSeminarCard
