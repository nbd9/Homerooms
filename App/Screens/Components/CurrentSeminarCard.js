// @flow

import React from 'react'
import { Text } from 'react-native'
import { Card, Button } from 'react-native-elements'
import Moment from 'moment'
import { Fonts, Colors } from '../../Themes'
import type { Teacher } from '../../Types/DatabaseTypes'

const CurrentSeminarCard = ({ day, seminarTeacher, onClick, icon = 'email', title = 'E-Mail Teacher', nextDay }: { day: string, seminarTeacher: Teacher, onClick: (email: string) => void, icon: string, title: string, nextDay: Moment }) => {
  let teacherPic = (seminarTeacher && 'picture' in seminarTeacher) ? {uri: seminarTeacher.picture} : null
  if (seminarTeacher) {
    return (
      <Card image={teacherPic}>
        {(nextDay)
          ? (
            <Text style={{ ...Fonts.style.content, marginBottom: 10 }}>
              Your next <Text style={{ ...Fonts.style.content, fontWeight: 'bold' }}>{day}</Text> day Homeroom is with <Text style={{ ...Fonts.style.content, fontWeight: 'bold' }}>{seminarTeacher.firstName} {seminarTeacher.lastName}</Text> in room <Text style={{ fontWeight: 'bold' }}>{seminarTeacher.room}</Text> on <Text style={{ fontWeight: 'bold' }}>{nextDay.format('dddd, MMMM Do')}</Text>
            </Text>
          )
          : <Text style={{ ...Fonts.style.content, marginBottom: 10 }}>
            Your <Text style={{ ...Fonts.style.content, fontWeight: 'bold' }}>{day}</Text> day Homerooms are with <Text style={{ ...Fonts.style.content, fontWeight: 'bold' }}>{seminarTeacher.firstName} {seminarTeacher.lastName}</Text>
          </Text>}

        <Button
          icon={{ name: icon }}
          backgroundColor={Colors.lightBlue}
          fontFamily={Fonts.type.headings}
          buttonStyle={{ borderRadius: 2, marginLeft: 0, marginRight: 0, marginBottom: 0 }}
          title={title}
          onPress={() => onClick(seminarTeacher.email)} />
      </Card>
    )
  }

  return null
}

export default CurrentSeminarCard
