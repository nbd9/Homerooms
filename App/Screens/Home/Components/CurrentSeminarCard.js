import React from 'react'
import { Text } from 'react-native'
import { Card, Button } from 'react-native-elements'
import { Fonts, Colors } from '../../../Themes'

const CurrentSeminarCard = ({ seminarTeacher }) => {
  if (seminarTeacher && 'picture' in seminarTeacher) {
    return (<Card
      image={{ uri: seminarTeacher.picture }}>
      <Text style={{ ...Fonts.style.content, marginBottom: 10 }}>
        Your next Support Seminar is with <Text style={{ ...Fonts.style.content, fontWeight: 'bold' }}>{seminarTeacher.firstName} {seminarTeacher.lastName}</Text> in room <Text style={{ fontWeight: 'bold' }}>{seminarTeacher.room}</Text>
      </Text>
      <Button
        icon={{ name: 'email' }}
        backgroundColor={Colors.lightBlue}
        fontFamily={Fonts.type.headings}
        buttonStyle={{ borderRadius: 0, marginLeft: 0, marginRight: 0, marginBottom: 0 }}
        title='E-Mail Teacher' />
    </Card>)
  } else if (seminarTeacher) {
    return (<Card>
      <Text style={{ ...Fonts.style.content, marginBottom: 10 }}>
        Your next Support Seminar is with <Text style={{ ...Fonts.style.content, fontWeight: 'bold' }}>{seminarTeacher.firstName} {seminarTeacher.lastName}</Text> in room <Text style={{ fontWeight: 'bold' }}>{seminarTeacher.room}</Text>
      </Text>
      <Button
        icon={{ name: 'email' }}
        backgroundColor={Colors.lightBlue}
        fontFamily={Fonts.type.headings}
        buttonStyle={{ borderRadius: 0, marginLeft: 0, marginRight: 0, marginBottom: 0 }}
        title='E-Mail Teacher' />
    </Card>)
  }

  return null
}

export default CurrentSeminarCard
