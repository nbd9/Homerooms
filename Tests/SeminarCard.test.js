import 'react-native'
import React from 'react'
import CurrentSeminarCard from '../App/Screens/Home/Components/CurrentSeminarCard'

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer'

test('renders correctly', () => {
  const testSeminarTeacher = {
    email: 'contact@nbdeg.com',
    firstName: 'Nick',
    id: '-L-9gHZsBoau-s4TYwSS',
    lastName: 'DeGroot',
    picture: 'https://avatars1.githubusercontent.com/u/196647...',
    room: 205,
    taughtCourses: 'AP Computer Science A'
  }
  const tree = renderer.create(<CurrentSeminarCard seminarTeacher={testSeminarTeacher} />).toJSON()
  expect(tree).toMatchSnapshot()
})
