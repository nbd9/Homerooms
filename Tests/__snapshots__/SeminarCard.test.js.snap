// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "white",
        "borderColor": "#e1e8ee",
        "borderWidth": 1,
        "margin": 15,
        "marginBottom": 0,
        "padding": 15,
        "shadowColor": "rgba(0,0,0, .2)",
        "shadowOffset": Object {
          "height": 0,
          "width": 0,
        },
        "shadowOpacity": 1,
        "shadowRadius": 1,
      },
      Object {
        "padding": 0,
      },
      undefined,
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
        },
        undefined,
        undefined,
      ]
    }
  >
    <View
      style={undefined}
    >
      <View
        style={
          Array [
            Object {
              "height": 150,
              "width": null,
            },
            undefined,
          ]
        }
      >
        <Image
          resizeMode="cover"
          source={
            Object {
              "uri": "https://avatars1.githubusercontent.com/u/196647...",
            }
          }
          style={
            Array [
              Object {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              },
              Object {
                "height": undefined,
                "width": undefined,
              },
              undefined,
            ]
          }
        />
      </View>
      <View
        style={
          Array [
            Object {
              "padding": 10,
            },
            undefined,
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "rgba(0, 0, 0, 0.87)",
              "fontFamily": "Nunito",
              "fontSize": 14,
              "lineHeight": 14,
              "marginBottom": 10,
            }
          }
        >
          Your 
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "rgba(0, 0, 0, 0.87)",
                "fontFamily": "Nunito",
                "fontSize": 14,
                "fontWeight": "bold",
                "lineHeight": 14,
              }
            }
          />
           day Support Seminar is with 
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "rgba(0, 0, 0, 0.87)",
                "fontFamily": "Nunito",
                "fontSize": 14,
                "fontWeight": "bold",
                "lineHeight": 14,
              }
            }
          >
            Nick
             
            DeGroot
          </Text>
           in room 
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "fontWeight": "bold",
              }
            }
          >
            205
          </Text>
        </Text>
        <View
          style={
            Array [
              Object {
                "marginLeft": 15,
                "marginRight": 15,
              },
              undefined,
              undefined,
              undefined,
            ]
          }
        >
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel={undefined}
            accessibilityTraits={undefined}
            accessible={true}
            hasTVPreferredFocus={undefined}
            hitSlop={undefined}
            isTVSelectable={true}
            nativeID={undefined}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                },
                undefined,
              ]
            }
            testID={undefined}
            tvParallaxProperties={undefined}
          >
            <View
              pointerEvents="box-only"
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "#9E9E9E",
                    "flexDirection": "row",
                    "justifyContent": "center",
                    "padding": 19,
                  },
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  Object {
                    "backgroundColor": "#1E88E5",
                  },
                  undefined,
                  Object {
                    "padding": 12,
                  },
                  undefined,
                  undefined,
                  undefined,
                  Object {
                    "borderRadius": 0,
                    "marginBottom": 0,
                    "marginLeft": 0,
                    "marginRight": 0,
                  },
                  undefined,
                  undefined,
                ]
              }
            >
              <Text
                accessible={true}
                allowFontScaling={false}
                ellipsizeMode="tail"
                style={
                  Array [
                    Object {
                      "color": "white",
                      "fontSize": 18,
                    },
                    Array [
                      Object {
                        "marginRight": 10,
                      },
                      undefined,
                    ],
                    Object {
                      "fontFamily": "Material Icons",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                  ]
                }
              >
                î‚¾
              </Text>
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Array [
                    Object {},
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    Array [
                      Object {
                        "color": "white",
                        "fontSize": 20,
                      },
                      undefined,
                      Object {
                        "fontSize": 17.5,
                      },
                      undefined,
                      undefined,
                      undefined,
                      Object {
                        "fontFamily": "Josefin Slab",
                      },
                    ],
                  ]
                }
              >
                E-Mail Teacher
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
